stew/result.nim

```diff
@@ -269,6 +279,32 @@ func `==`*(lhs, rhs: Result): bool {.inline.} =
 func get*[T: not void, E](self: Result[T, E]): T {.inline.} =
   ## Fetch value of result if set, or raise error as an Exception
   ## See also: Option.get
+  ##
+  ## TODO `raiseResultError` is considered a CatchableError which
```

Comment:
I consider this a serious design issue that needs addressing.

Reply:
If we agree to raise a `Defect` here, I think the `toException` feature doesn't make much sense anymore in this particular context. If a `tryGet` function is added, then it may be used to turn the recoverable error value into a recoverable exception.

Reply:
yeah, let's do `Defect`.

re `toException`, maybe there is utility still - it gives the user of the library the possibility to customize `Result` for their type as a way to easily transition to the catchable Exception world and it's opt-in. I imagine that if I was converting an API from exceptions to Result, I'd include a backwards-compatible API implemented with `toException` - a little like `boost asio` offers both exception and non-exception-based API. 


One thing that's missing in nim in generalare push and pull conversion feature between error types (enums of different categories) - rust has its general purpose conversion framework with the to/from traits while c++ has constructors and type operators - these help keep the boilerplate down somewhat - `toException` is a special case here but what I'd really want the library to offer is some simple conversions that can be used at abstraction boundaries.

Reply:
so playing around with this a bit more, using `doAssert` for the defect means the defect is not tracked - I have a local version that raises a `ResultDefect` instead (with the error in it, if it's printable).. not sure what will happen with the suggested panic mode though (vs sysfatal)